core/tee/tee_svc_cryp.c

```diff
@@ -1585,6 +1585,8 @@ static TEE_Result tee_svc_cryp_obj_populate_type(
 				    obj_type == TEE_TYPE_DES3)
 					obj_size -= obj_size / 8;
 			}
+			if (obj_size > o->info.maxKeySize)
+				return TEE_ERROR_BAD_STATE;
```

Comment:
(for commit "core: check max key size when populating object")

From my reading of the specs, I think this should return `TEE_ERROR_BAD_PARAMETERS`.
> **5.6.4 TEE_PopulateTransientObject**
**Return Code**
• TEE_ERROR_BAD_PARAMETERS : If an incorrect or inconsistent attribute value is detected. In this case,
the content of the object SHALL remain uninitialized.

I think this comment also applies to return values used in commit "core: check that certain object attributes are consistent with max key size".

Reply:
Yes, but then there's the panic reason:
> *  An attribute value is too big to fit within the maximum object size specified when the object was created.